BraveWallet/Panels/Connect/EditSiteConnectionView.swift

```diff
@@ -8,50 +8,129 @@ import BraveCore
+      case .disconnect:
+        return Strings.Wallet.editSiteConnectionAccountActionDisconnect
+      case .switch:
+        return Strings.Wallet.editSiteConnectionAccountActionSwitch
+      }
+    }
+  }
   
-  private func actionTitle(for account: BraveWallet.AccountInfo) -> String {
-    // Disconnect - Connected and selected account
-    // Connect - Not connected
-    // Switch - Connected but not selected account
-    if keyringStore.selectedAccount.id == account.id {
-      return "Disconnect"
+  private func editAction(for account: BraveWallet.AccountInfo) -> EditAction {
+    if permittedAccounts.contains(account.address) {
+      if keyringStore.selectedAccount.id == account.id {
+        // Disconnect - Connected and selected account
+        return .disconnect
+      } else {
+        // Switch - Connected but not selected account
+        return .`switch`
+      }
+    } else {
+      // Connect - Not connected
+      return .connect
+    }
+  }
+  
+  @ViewBuilder private func editButton(action: EditAction, account: BraveWallet.AccountInfo) -> some View {
+    Button {
+      switch action {
+      case .connect:
+        if let url = origin.url {
+          Domain.setEthereumPermissions(forUrl: url, account: account.address, grant: true)
+        }
+        permittedAccounts.append(account.address)
+        keyringStore.selectedAccount = account
+      case .disconnect:
+        if let url = origin.url {
+          Domain.setEthereumPermissions(forUrl: url, account: account.address, grant: false)
+        }
+        permittedAccounts.removeAll(where: { $0 == account.address })
+        
+        if let firstAllowedAdd = permittedAccounts.first, let firstAllowedAccount = keyringStore.keyring.accountInfos.first(where: { $0.id == firstAllowedAdd }) {
+          keyringStore.selectedAccount = firstAllowedAccount
+        }
+      case .switch:
+        keyringStore.selectedAccount = account
+      }
+    } label: {
+      Text(action.title)
+        .foregroundColor(Color(.braveBlurpleTint))
+        .font(.footnote.weight(.semibold))
+    }
+  }
+  
+  @ViewBuilder private var connectionInfo: some View {
+    Image(systemName: "globe")
+      .frame(width: min(faviconSize, maxFaviconSize), height: min(faviconSize, maxFaviconSize))
+      .background(Color(.braveDisabled))
+      .clipShape(RoundedRectangle(cornerRadius: 10, style: .continuous))
+    VStack(alignment: sizeCategory.isAccessibilityCategory ? .center : .leading, spacing: 2) {
+      origin.url.map { url in
+        Text(verbatim: url.absoluteString)
+          .font(.subheadline.weight(.semibold))
+          .multilineTextAlignment(.center)
+          .foregroundColor(Color(.bravePrimary))
+      }
+      Text(String.localizedStringWithFormat(Strings.Wallet.editSiteConnectionConnectedAccount, permittedAccounts.count, permittedAccounts.count == 1 ? Strings.Wallet.editSiteConnectionAccountSingular : Strings.Wallet.editSiteConnectionAccountPlural))
```

Comment:
Minor but [a comment left](https://github.com/brave/brave-ios/pull/5347#discussion_r875693937) on one of my prior PRs was similar to this line to keep it to 120-140 line length if possible. I moved this line to a computed property and I think it helped a lot: https://github.com/brave/brave-ios/pull/5347/commits/858db2323564e3d7f02d4f66ed2162e8650d6a80#diff-e1ee7e6c88c5332e5b9da1f791d550f922eaad4e480574422ffb1025faf1656dR101